tests/integration_test/jwt_auth_integration_test.go

```diff
@@ -125,6 +125,15 @@ func TestAsymmetricKeys(t *testing.T) {
 			token:          testdata.FakeNonexistJwksProviderToken,
 			wantError:      `401 Unauthorized, {"code":401,"message":"Jwks remote fetch is failed"}`,
 		},
+		// TODO(b/179038530): should return `JwtUnknownIssuer`
+		{
+			desc:           "Failed, misleading error message(JWT is missing) on Jwt signed by an unknown issue",
+			clientProtocol: "http",
+			httpMethod:     "GET",
+			method:         "/v1/shelves?key=api-key",
+			token:          testdata.FakeUnknownIssuerToken,
+			wantError:      `401 Unauthorized, {"code":401,"message":"Jwt is missing"}`,
```

Comment:
So you can reproduce this error? 

How is the envoy jwt_authn config look like?

Reply:
Yes. 


https://paste.googleplex.com/6658566795558912

Reply:
Cool.   You found the problem.